### ISSUE TITLE
Inherited Pause in screenpy-selenium has errors in mypy

### ISSUE THREAD

**Issue by MarcelWilson** [ScreenPyHQ/screenpy#25]
screenp-selenium has a `Pause` class that inherits screenpy.actions.pause.Pause. 

mypy will error on the following code from screenpy-selenium
```python
Pause.for_(length).seconds_because("... reasons").add_to_chain(tester, chain)
```

`"Pause" has not attribute "add_to_chain" [attr-defined]`

**Comment by MarcelWilson** [comment]
The problem exists for any baseclass that annotates itself as the object returned by a method.  
The easy solution to this would be to create `TypeVar`s for each baseclass that would potentially be inherited.

**Comment by MarcelWilson** [comment]
See [Mypy docs](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#the-type-of-class-objects) for more details about how `TypeVar` is used.

**Comment by MarcelWilson** [comment]
[PEP 673](https://peps.python.org/pep-0673) was accepted and is included in 3.11 but  mypy still hasn't merged the update to support checking it.  I'm proposing we add `TypeVar` now and when https://github.com/python/mypy/pull/11666 eventually releases I'll update the package to utilize `Self`

**Comment by perrygoy** [comment]
Thanks to @bandophahita's hard work, this issue has been resolved!
